nix/docker.nix

```diff
@@ -1,12 +1,18 @@
 {
+  stdenv,
   dockerTools,
   cacert,
   text-generation-inference,
+  runCommand,
   stream ? false,
 }:
 
 let
   build = if stream then dockerTools.streamLayeredImage else dockerTools.buildLayeredImage;
+  tmp = runCommand "tmp" { } ''
+    mkdir $out
+    mkdir -m 1777 $out/tmp
+  '';
```

Comment:
I don't think an intermediate derivation is needed, `buildLayeredImage` takes an `extraCommands` argument where you can execute some additional commands as part of the last layer. See e.g. https://github.com/NixOS/nixpkgs/blob/1c85b7d96eb38b6b953eb08800f372a2eab624b1/pkgs/build-support/docker/examples.nix#L62 .